schemas/metadata-events/SetDataSchema.json

```diff
@@ -4,14 +4,19 @@
   "description": "Specifies the complete schema of Data Slices added to the Dataset following this event.",
   "type": "object",
   "additionalProperties": false,
-  "required": [
-    "schema"
-  ],
+  "required": [],
   "properties": {
-    "schema": {
+    "rawArrowSchema": {
+      "tag": 0,
       "type": "string",
       "format": "flatbuffers",
-      "description": "Apache Arrow schema encoded in its native flatbuffers representation."
+      "deprecated": true,
+      "description": "DEPRECATED: Apache Arrow schema encoded in its native flatbuffers representation."
+    },
+    "schema": {
+      "tag": 1,
+      "$ref": "/schemas/DataSchema",
+      "description": "Defines the logical schema of the data files that follow this event. Will become a required field after migration."
```

Comment:
I don't fully understand how is this backward compatible, if the old field that exists in the wild now is called differently.

Reply:
RFC-16 Compatibility section explains it. Briefly:
- Flatbuffer compatibility is achieved via tags (field IDs) as FB does not use names
- In YAML manifests `SetDataSchema` was never specified before, so we are not going to break any dataset snapshots by renaming the field